File: Assignment No_16/Question_5.c
/*5. Write a program which accept N numbers from the user and return product of all odd elements.

Input : 15 66 3 70 10 88
Output : 45

Input : 44 66 72 70 10 88
Output : 0


*/
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>
#include<stdlib.h>

int Product_Odd(int Arr[], int iSize)
{
  int iCnt=0, iProduct=1;

  for(iCnt=0; iCnt<iSize; iCnt++)
  {
    if(Arr[iCnt] % 2 !=0)
      iProduct= iProduct*Arr[iCnt];
  }

  if(iProduct == 1)
    iProduct =0;

  return iProduct;
 
}


int main()
{
   int iSize=0, iCnt=0;
   int *Array=NULL;
   int iRet=0;

   printf("Enter the Number of Elements:\t");
   scanf("%d",&iSize);

   Array=(int *)malloc(iSize*sizeof(int));

   if(Array==NULL)
     return -1;


   printf("Enter %d Elements",iSize);
   for(iCnt=0; iCnt<iSize; iCnt++)
     scanf("%d",&Array[iCnt]);
   
   
   iRet=Product_Odd(Array,iSize);
  
   printf("Product of Odd Elements:%d\t",iRet);

   free(Array);

   return 0;
}

File: Assignment No_39/Question_5.java
 /* 5. Write java program which accept N numbers from user and return product of all odd elements.

Input : N : 6
Elements :15 66 3 70 10 88       (15*3)
Output : 45

Input : N : 6
Elements :44 66 72 70 10 88      (No Odd Element Present)
Output : 0

*/
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

 
 import java.util.*;

class Ass39_5
{
    public static int Product_Of_Odd_Elements(int Brr[])
    {
        int i = 0, iProduct=1;

        for(i = 0; i < Brr.length; i++)
        {
            if(Brr[i] % 2 != 0)
            {
                iProduct = iProduct*Brr[i];
            }
        }
       
       if(iProduct > 1)
        return iProduct;
       else 
        return 0;
        
    }

    public static void main(String Arg[])
    {
        Scanner sobj = new Scanner(System.in); 
        int i = 0, iSize = 0;
        int iRet = 0;

        System.out.println("Enter the number of elements : ");
        iSize = sobj.nextInt();

        int Arr[] = new int[iSize];

        System.out.println("Enter the values : ");

        for(i = 0; i < iSize; i++)
        {
            Arr[i] = sobj.nextInt();
        }

        System.out.println("Elements of the array are : ");

        for(i = 0; i < iSize; i++)
        {
            System.out.print(Arr[i]+"\t");
        }
        System.out.println();


        iRet=Product_Of_Odd_Elements(Arr);

        System.out.println(+iRet);

    }
}



